src/app/view/dialogs/trainrun-and-section-dialog/trainrunsection-card/trainrun-section-card.component.ts

```diff
@@ -54,41 +52,56 @@ export class TrainrunSectionCardComponent implements AfterViewInit, OnDestroy {
     this.trainrunSectionService.trainrunSections.pipe(takeUntil(this.destroyed)).subscribe(() => {
       this.updateAllValues();
     });
+  }
+
+  ngOnInit() {
+    this.updateAllValues();
+  }
+
+  updateAllValues() {
+    const selectedTrainrun = this.trainrunService.getSelectedTrainrun();
+    if (!selectedTrainrun) {
+      return;
+    }
+    const bothEndNodes = this.trainrunService.getBothEndNodesWithTrainrunId(
+      selectedTrainrun.getId(),
+    );
+    const node = GeneralViewFunctions.getLeftOrTopNode(
+      bothEndNodes.endNode1,
+      bothEndNodes.endNode2,
+    );
+    const trainrunSection = node.getStartTrainrunSection(selectedTrainrun.getId());
+    if (!trainrunSection) {
+      return;
+    }
```

Comment:
I would suggest use an alternative way of finding the first trainrun section of a trainrun, that is now possible due to the `source->target` chain guarantee (simplify the code and tends to reduce the manipulations needed to navigate in a trainrun)

What do you think @emersion?

```suggestion
  updateAllValues() {
    const selectedTrainrun = this.trainrunService.getSelectedTrainrun();
    if (!selectedTrainrun) {
      return;
    }
    const trainrunSection = this.trainrunService.getFirstTrainrunSection(selectedTrainrun);
```